src/transform/parameters.ts

```diff
@@ -16,9 +17,10 @@ export function transformParametersArray(
       if (globalParams[paramName]) {
         const reference = globalParams[paramName] as any;
         if (!mappedParams[reference.in]) mappedParams[reference.in] = {};
+        const schema = { $ref: paramObj.$ref };
         mappedParams[reference.in][reference.name || paramName] = {
           ...reference,
-          schema: { $ref: paramObj.$ref },
+          ...(version === 2 ? schema : version === 3 ? { schema } : {}),
```

Comment:
So because this is a bit difficult to follow, let’s take up a little more space:

```ts
if (version === 2) {
  mappedParams[reference.in][reference.name || paramName] = {
    ..reference,
    $ref: paramObj.$ref,
  } as any;
} else if (version === 3) {
  mappedParams[reference.in][reference.name || paramName] = {
    ..reference,
    schema: { $ref: paramObj.$ref },
  } as any;
}
```

Your version takes up fewer lines, and is good in that regard. But you have a nested ternary + a spread operator, and it’s very easy for someone to make a mistake (if not you, the next person to touch this).

Reply:
I agree with you. Fixed in https://github.com/drwpow/openapi-typescript/pull/489/commits/8c02bd78cbc0ac13d83632ff358991056c54bfa2

Perhaps we need to consider the case where `in` is `body` here as well??